### PR TITLE
Readme for optimistic locking and events versioning strategies

### PR DESCRIPTION
**Pull request by pienikrz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,16 +193,133 @@ Let's now create an instance of `OrderRepository` passing previously created eve
...
 repository.save(new OrderPlaced(UUID.randomUUID(), aggregateId, "PizzaOrder3214"));
-repository.save(new OrderCancelled(UUID.randomUUID(), aggregateId, "I'm not hungry anymory"));
+repository.save(new OrderCancelled(UUID.randomUUID(), aggregateId, "I'm not hungry anymore"));
 Optional<Order> order = repository.find(aggregateId);
 ```
 
-## Events versioning <a name="EventsVersioning"></a>
+## Optimistic locking <a name="OptimisticLocking"></a>
+
+Let's assume that you have `Order` aggregate in your code with the rule that if order has been sent it can't be cancel anymore.
```
Comment (rydlu) [inline README.md:202]: cancelled

File: README.md [hunk 2]
```diff
@@ -193,16 +193,133 @@ Let's now create an instance of `OrderRepository` passing previously created eve
...
+
+    Order applyVersion(Integer version) {
+        this.version = version;
+        return this;
+    }
+
+}
+```
+
+Now we have to create repository for `Order` aggregate
```
Comment (rydlu) [inline README.md:254]: a repository

**Review by rydlu — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -193,16 +193,133 @@ Let's now create an instance of `OrderRepository` passing previously created eve
...
+                INITIAL_AGGREGATE_STATE_SUPPLIER,
+                AGGREGATE_EVENT_APPLIER,
+                AGGREGATE_VERSION_APPLIER
+        );
+    }
+
+}
+```
+
+Please note that in additional to repository created in [dealing with aggregates](#DealingWithAggregates) chapter there is additional parameter AGGREGATE_VERSION_APPLIER.
```
Comment (michalkarolik) [inline README.md:277]: in addition to

File: README.md [hunk 4]
```diff
@@ -6,41 +6,47 @@
...
 ## How to add it to project <a name="HowToAddItToProject"></a>
 
 ### Add dependency
 
-If you are using gradle add it to build.gradle:
+If you are using gradle add this to build.gradle:
 ```groovy
 dependencies {
-    implementation "com.hltech:sql-event-store-4j:version"
+    implementation "com.hltech:sql-event-store-4j:0.0.7"
```
Comment (michalkarolik) [inline README.md:33]: maybe lets stick to 'version', we wont have to upgrade this piece of readme on each release

**Review by michalkarolik — APPROVED** [review]

**Pull request merged**